### ISSUE TITLE
Feature Request: have onRate fire on mouse up instead of mouse down

### ISSUE THREAD

**Issue by bslinger** [dreyescat/react-rating#25]
Not sure if this would cause any other problems, I can attempt a PR if you like.

The reason I would like it is that I'm working on a project where we show a popover when a user rates something, but the mouseDown is triggering the rating and the mouseUp is being caught by the (now visible) popover and closing it.

I feel like it is more in line with what I would expect anyway - if I mouse down and then change my mind about the rating I want to give, dragging to the correct rating and doing a mouse up then would fire the desired rating.

Let me know your thoughts :)

**Comment by dreyescat** [comment]
I took a quick look at the code and right now I don't remeber why I decided to use the `onMouseDown` as the event triggering the `onChange(rate)`. Even the `onClick(rate)` event too!.

What you say makes sense to me. However, instead of relying on the `onMouseUp`, why not jus use the `onClick` event instead? This is the event that is expected to be fired when you [press and release on a single element](https://developer.mozilla.org/en-US/docs/Web/Events/click).

I propose something like this:

```bash
grep -rl 'MouseDown' src/ | xargs sed -i 's/MouseDown/Click/g'
```

That is, replace every `onMouseDown` occurrence by `onClick`.

Does it make sense to you? Could you try your popover scenario with these changes and see if it works?

**Comment by dreyescat** [comment]
Great!

I released and published a new version [v0.6.1](https://github.com/dreyescat/react-rating/releases/tag/v0.6.1) with your changes.
